### PR TITLE
feat: move pindel region bed files to input

### PR DESCRIPTION
**Pull request by maehler**
### Description

Recently I discovered that if you pass a non-existent bed file as a value for the command line parameters `-j/--include` or `-J/--exclude` in pindel, the program behaves as if empty files are being passed. Since these are supplied in `params.extra`, no checks are done on them.

I realise that this is not an issue with the wrapper, per se, but the pindel repo has been archived, and I think that modifying the wrapper might be a good way to catch potential mistakes.

Not sure whether my implementation is the best way to handle things, so thoughts and opinions are welcome.

### QC

* [x] I confirm that:

For all wrappers added by this PR, 

* there is a test case which covers any introduced changes,
* `input:` and `output:` file paths in the resulting rule can be changed arbitrarily,
* either the wrapper can only use a single core, or the example rule contains a `threads: x` statement with `x` being a reasonable default,
* rule names in the test case are in [snake_case](https://en.wikipedia.org/wiki/Snake_case) and somehow tell what the rule is about or match the tools purpose or name (e.g., `map_reads` for a step that maps reads),
* all `environment.yaml` specifications follow [the respective best practices](https://stackoverflow.com/a/64594513/2352071),
* wherever possible, command line arguments are inferred and set automatically (e.g. based on file extensions in `input:` or `output:`),
* all fields of the example rules in the `Snakefile`s and their entries are explained via comments (`input:`/`output:`/`params:` etc.),
* `stderr` and/or `stdout` are logged correctly (`log:`), depending on the wrapped tool,
* temporary files are either written to a unique hidden folder in the working directory, or (better) stored where the Python function `tempfile.gettempdir()` points to (see [here](https://docs.python.org/3/library/tempfile.html#tempfile.gettempdir); this also means that using any Python `tempfile` default behavior works),
* the `meta.yaml` contains a link to the documentation of the respective tool or command,
* `Snakefile`s pass the linting (`snakemake --lint`),
* `Snakefile`s are formatted with [snakefmt](https://github.com/snakemake/snakefmt),
* Python wrapper scripts are formatted with [black](https://black.readthedocs.io).
* Conda environments use a minimal amount of channels, in recommended ordering. E.g. for bioconda, use (conda-forge, bioconda, nodefaults, as conda-forge should have highest priority and defaults channels are usually not needed because most packages are in conda-forge nowadays).

### CODE REVIEW

**Comment by fgvieira** [comment]
Why not just have the BED file as an input file? This way `snakemake` will check if the file exists.

**Comment by maehler** [comment]
I was thinking to have the bed files as inputs, but that more or less also makes them required, which I kind of wanted to avoid. If I recall correctly, it is possible to set an input to an empty list for it to not fail, and that could then be handled by the wrapper, but that's also kind of ugly. What do you think?

Good point about the prefix. I just followed the existing wrapper blindly. Having the prefix managed by the wrapper would be much cleaner.

**Comment by maehler** [comment]
> I was thinking to have the bed files as inputs, but that more or less also makes them required, which I kind of wanted to avoid. If I recall correctly, it is possible to set an input to an empty list for it to not fail, and that could then be handled by the wrapper, but that's also kind of ugly.

Or no, I'm being a square. Whether the argument would be optional or not would be handled by the wrapper. Much cleaner.

**Comment by fgvieira** [comment]
Just modify this one...

File: bio/pindel/call/wrapper.py [hunk 1]
```diff
@@ -3,13 +3,34 @@
...
+if include_bed is not None and exclude_bed is not None:
+    raise Exception("supply either include_bed or exclude_bed, not both")
+
+include_arg = ""
+if include_bed is not None:
+    include_arg = "-j {}".format(include_bed)
+
+exclude_arg = ""
+if exclude_bed is not None:
+    exclude_arg = "-J {}".format(exclude_bed)
```
Comment (fgvieira) [inline bio/pindel/call/wrapper.py:23]:
```suggestion
    exclude_arg = f"-J {exclude_bed}"
```

File: bio/pindel/call/wrapper.py [hunk 2]
```diff
@@ -3,13 +3,34 @@
...
 
+include_bed = snakemake.input.get("include_bed", None)
+exclude_bed = snakemake.input.get("exclude_bed", None)
+
+if include_bed is not None and exclude_bed is not None:
+    raise Exception("supply either include_bed or exclude_bed, not both")
+
+include_arg = ""
+if include_bed is not None:
+    include_arg = "-j {}".format(include_bed)
```
Comment (fgvieira) [inline bio/pindel/call/wrapper.py:19]:
```suggestion
    include_arg = f"-j {include_bed}"
```

**Comment by fgvieira** [comment]
It looks great! Just some small cosmetic suggestions. :smile:

**Review by fgvieira — APPROVED** [review]

**Comment by maehler** [comment]
> It looks great! Just some small cosmetic suggestions. 😄

Great, thanks for the input! This turned out a lot better.

**Comment by fgvieira** [comment]
It seems there is a conflict on `test.py`. Can you take a look at it?

**Comment by maehler** [comment]
Should be good to go now.

**Review by fgvieira — APPROVED** [review]

**Pull request merged**